Automattic/woocommerce-payments
# Allow the `process-changelog` composite action to proceed even without any significant changes

Fixes #5230 

#### Changes proposed in this Pull Request

- Instead of returning an exit code when no changes are found (either no changelog files or only insignificant changes), Jetpack Changelogger now returns a warning because we provide the `--yes` flag

This allows the GitHub workflows using the `process-changelog` composite action to continue their execution and not fail at this step.
The only output added to `changelog.txt` and `readme.txt` will be the release version and release date because that's all Jetpack can output given the lack of significant changelog entries:
```
= x.y.z - YYYY-MM-DD =
```

With this, we can have a proper patch release workflow and stop asking to merge the fixes on `trunk` first, before creating the release branch from it.

We'll be able to:
- Create a release branch from `trunk` using the [Release - Create PR to trunk](https://github.com/Automattic/woocommerce-payments/actions/workflows/release-pr.yml) workflow (which will not fail anymore due to lack of changelog files on `trunk`)
- Ask that all fixes be prepared on individual `fix/...` branches and merged/cherry-picked into the release branch directly (instead of `trunk` like today)
- Amend the changelog entries using the [Release - Changelog](https://github.com/Automattic/woocommerce-payments/actions/workflows/release-changelog.yml)

#### Testing instructions

On any branch without any changelog files under `changelog`, or with insignificant changelog files only (choose `patch` and then confirm the insignificant entry when asked, i.e. leave empty your response):
- Run the [Release - Changelog](https://github.com/Automattic/woocommerce-payments/actions/workflows/release-changelog.yml) to **generate** entries. It should only add the release version and date to `changelog.txt` and `readme.txt`
- Add some changelog entries (significant ones) to the branch
- Run again the [Release - Changelog](https://github.com/Automattic/woocommerce-payments/actions/workflows/release-changelog.yml) to **amend** entries. It should update the `changelog.txt` and `readme.txt` files with the changelog entries you previously pushed

This simulates how the patch release process would look in the future, as I described above (first generate changelog even though there are no entries to generate from, then add changelog entries, and finally amend the changelog)

-------------------


**Post merge**
